c_src/brotli_nif.c

```diff
@@ -215,60 +216,60 @@ static ERL_NIF_TERM brotli_encoder_compress_stream(ErlNifEnv *env, int argc,
     return BADARG;
   }
 
-  size_t available_out = 0;
+
   size_t available_in = input.size;
   const uint8_t *next_in = input.data;
+  size_t available_out = 0;
+  uint8_t *next_out = NULL;
+  BROTLI_BOOL ok = BROTLI_TRUE;
 
-  return EBOOL(BrotliEncoderCompressStream(*state, op, &available_in, &next_in,
-                                           &available_out, NULL, NULL));
-}
+  if (!enif_alloc_binary(0, &output)) {
+    return atom_error;
+  }
 
-static ERL_NIF_TERM brotli_encoder_has_more_output(ErlNifEnv *env, int argc,
-                                                   const ERL_NIF_TERM argv[]) {
-  assert_argc(1);
-  BrotliEncoderState **state;
+  while (ok) {
+    ok = BrotliEncoderCompressStream(*state, op,
+                                     &available_in, &next_in,
+                                     &available_out, &next_out, NULL);
+    if (!ok)
+      break;
 
-  if (!enif_get_resource(env, argv[0], encoder_state_resource_type,
-                         (void **)&state)) {
-    return BADARG;
-  }
+    size_t output_chunk_size = 0; // Request all available output.
+    const uint8_t* data = BrotliEncoderTakeOutput(*state, &output_chunk_size);
```

Comment:
`BrotliEncoderTakeOutput` is not guaranteed to return all the available output - [link](https://brotli.org/encode.html#afd6)
Probably we should loop while `BrotliEncoderHasMoreOutput = true`